improver/psychrometric_calculations/total_precipitable_water.py

```diff
@@ -0,0 +1,169 @@
+    Pascals and uses constants for gravity and water density to convert the result
+    into liquid water equivalent thickness.
+
+    Attributes:
+        model_type (str): Type of model used, e.g., 'global' or 'ukv'.
+    """
+
+    def __init__(self, model_type: str = "global") -> None:
+        """
+        Initialise the PrecipitableWater plugin.
+
+        Args:
+            model_type:
+                Type of model used, e.g., 'global' or 'ukv'.
+        """
+        self.model_type = model_type.lower()
+
+    @staticmethod
+    def calculate_pressure_thickness(
+        layer_bounds: ndarray, cube_data: ndarray
+    ) -> ndarray:
+        """
+        Calculate the pressure thickness of each atmospheric layer.
+
+        Args:
+            layer_bounds:
+                1D array of pressure layer boundaries in Pascals.
+            cube_data:
+                N-dimensional array of humidity mixing ratio values.
+
+        Returns:
+            N-dimensional array of pressure thickness values reshaped to match
+            the dimensions of the input data.
+        """
+        delta_p = -np.diff(layer_bounds)
+        reshaped_delta_p = delta_p.reshape(
+            (len(delta_p),) + (1,) * (cube_data.ndim - 1)
+        )
+        return reshaped_delta_p
+
+    @staticmethod
+    def calculate_precipitable_water(
+        sorted_data: ndarray, reshaped_delta_p: ndarray
+    ) -> ndarray:
+        """
+        Calculate total precipitable water using sorted humidity data and pressure thickness.
+
+        Args:
+            sorted_data:
+                Humidity mixing ratio data sorted by pressure levels.
+            reshaped_delta_p:
+                Pressure thickness values reshaped to match the data dimensions.
+
+        Returns:
+            Array of total precipitable water values in metres.
+        """
+        pw_data = (
+            sorted_data
+            * reshaped_delta_p
+            / (EARTH_SURFACE_GRAVITY_ACCELERATION * WATER_DENSITY)
+        )
+        pw_data = np.maximum(pw_data, 0)
+        total_pw_data = np.sum(pw_data, axis=0)
+        return total_pw_data
+
+    def sort_pressure_levels(self, cube: Cube) -> Tuple[ndarray, ndarray]:
+        """
+        Sort pressure levels in descending order and reorder the data accordingly.
+
+        Args:
+            cube:
+                Cube containing humidity mixing ratio data with a pressure coordinate.
+
+        Returns:
+            Tuple of sorted pressure levels and corresponding sorted data.
+
+        Raises:
+            ValueError: If the pressure coordinate is missing or not in Pascals.
+        """
+        try:
+            pressure_coord = cube.coord("pressure")
+        except iris.exceptions.CoordinateNotFoundError:
+            raise ValueError("Cube must have a 'pressure' coordinate.")
+
+        if pressure_coord.units != "Pa":
+            raise ValueError("Pressure units must be in Pascals (Pa).")
```

Comment:
I don't think there's a unit test(s) for these exceptions.

Also, you could replace the second one with `pressure_coord.convert_units("Pa")`, which will raise an exception if this isn't possible, but will also convert from `hPa`, which might be useful.